### PR TITLE
fix for fix

### PR DESCRIPTION
**Pull request by alamb**
Proposed modification for https://github.com/apache/arrow-rs/pull/236

### CODE REVIEW

File: arrow/src/compute/kernels/sort.rs [hunk 1]
```diff
@@ -1442,7 +1444,7 @@ mod tests {
                 nulls_first: true,
             }),
             Some(3),
-            vec![None, None, Some(2)],
+            vec![None, Some(2), Some(0)],
```
Comment (alamb) [inline arrow/src/compute/kernels/sort.rs:1447]: I think the checked in answers are incorrect -- namely the first three items in the input are `None`, `0` and `2` and yet the output has two nulls in it.

File: arrow/src/compute/kernels/sort.rs [hunk 2]
```diff
@@ -480,18 +480,20 @@ where
...
 
     let valids_len = valids.len();
-    let nulls_len = nulls.len();
+    let mut nulls_len = nulls.len();
     let mut len = values.len();
 
     if let Some(limit) = limit {
         len = limit.min(len);
+        // count how many nulls are present in the limit
+        nulls_len = nulls.iter().filter(|&idx| *idx as usize <= limit).count();
```
Comment (medwards) [inline arrow/src/compute/kernels/sort.rs:489]: Everything else in this function doesn't need to dig into the actual index values so something smells here to me. I thought the limit is applied to the sorted result so I don't know that comparing the limit to the indices is correct here.

File: arrow/src/compute/kernels/sort.rs (hunk 1 above)
Comment (medwards) [reply arrow/src/compute/kernels/sort.rs:1447]: Is this passing? This looks wrong to me, the unoptimized order (ie /wo limit) would look like `None, None, 2, 0, 0, -1`, applying the limit: `None, None, 2`.

**Comment by alamb** [comment]
I misunderstood the expected behavior. Sorry about that

**Pull request closed without merging**